src/hooks/useTankMapping.ts

```
@@ -0,0 +1,214 @@
+
+export function useTankMapping(): UseTankMappingReturn {
+  const { tankTables } = useTankTables();
+  const [mappings, setMappings] = useState<TankMapping[]>([]);
+  const [mappingConfiguration, setMappingConfiguration] = useState<TankMappingConfiguration | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isGeneratingSuggestions, setIsGeneratingSuggestions] = useState(false);
+
+  // Load mappings and configuration on mount
+  useEffect(() => {
+    loadMappingData();
+  }, [loadMappingData]);
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)